kqueen_ui/blueprints/ui/views.py

```diff
@@ -700,17 +700,83 @@ def handle(self, cluster_id):
         cluster = self.kqueen_request('cluster', 'get', fnargs=(cluster_id,))
         if 'node_count' not in cluster.get('metadata', {}):
             engine = cluster.get('provisioner', {}).get('engine', '<unknown>')
-            flash("{} engine doesn't support scaling.".format(prettify_engine_name(engine)), 'warning')
-            return redirect(request.environ.get('HTTP_REFERER', url_for('ui.index')))
+            flash("{} engine doesn't support scaling."
+                  .format(prettify_engine_name(engine)), 'warning')
+            return redirect(request.environ.get('HTTP_REFERER',
+                                                url_for('ui.index')))
         current_node_count = cluster['metadata']['node_count']
         node_count = request.form['node_count']
+
+        if int(node_count) < 2 and cluster['networkPolicy']['enabled'] is True:
```

Comment:
I would still prefer to create a way to check available features through API call to backend, but if you want to check in UI, you still need to take into account that not all clusters have `networkPolicy` key, so it should be - `cluster.get('networkPolicy', {}).get('enabled', False)` instead. Also I thought its under `metadata` key, not a top level attribute of cluster? If so then its - `cluster.get('metadata', {}).get('networkPolicy', {}).get('enabled', False)`